currency.go

```diff
@@ -0,0 +1,59 @@
+package money
+
+import (
+	"errors"
+	"strings"
+)
+
+type Currency struct {
```

Comment:
maybe we should add an interface of Currency to allow customization on the method implementation